src/OMSimulatorLib/Model.cpp

```diff
@@ -216,8 +216,9 @@ oms_status_enu_t oms::Model::importSnapshot(const char* snapshot_, char** newCre
   ComRef new_cref = ComRef(ssdNode.attribute("name").as_string());
   std::string ssdVersion = ssdNode.attribute("version").as_string();
 
-  if (new_cref != getCref())
-    return logError("this API cannot be used to rename a model");
+  // check new_cref exists in scope
+  if (new_cref != getCref() && Scope::GetInstance().getModel(new_cref))
+    return logError("renaming model name \"" + std::string(new_cref) + "\" failed as it already exists in the scope");
```

Comment:
return logError("Renaming the model \"" + std::string(getCref()) + "\" to \"" + std::string(new_cref) + "\" failed because another model with the same name already exists in the scope.");

Reply:
fixed with https://github.com/OpenModelica/OMSimulator/pull/1006/commits/37e754c71c844976f3f1cceed2d1ec20cf15d383